src/Features/LanguageServer/Protocol/Handler/Diagnostics/WorkspacePullDiagnosticHandler.cs

```diff
@@ -56,71 +58,124 @@ protected override DiagnosticTag[] ConvertTags(DiagnosticData diagnosticData)
             return ConvertTags(diagnosticData, potentialDuplicate: true);
         }
 
-        protected override ValueTask<ImmutableArray<IDiagnosticSource>> GetOrderedDiagnosticSourcesAsync(RequestContext context, CancellationToken cancellationToken)
-            => GetDiagnosticSourcesAsync(context, GlobalOptions, cancellationToken);
+        protected override async ValueTask<ImmutableArray<IDiagnosticSource>> GetOrderedDiagnosticSourcesAsync(
+            VSInternalWorkspaceDiagnosticsParams diagnosticsParams,
+            RequestContext context,
+            CancellationToken cancellationToken)
```

Comment:
helper to walk projects in priority order.  used by both teh normal diagnostic walking code, and the task-list code.